src/components/DERP/index.jsx

```diff
@@ -1,33 +1,101 @@
-import React, { useState, useEffect } from "react";
+import React, { useState, useEffect, useRef } from "react";
 // import CheckCircleIcon from "@mui/icons-material/CheckCircle";
 
 import { chains } from "../../shared/chains.js";
 import Map from "../Map";
-import { Location } from "./styled"
-import DERPLog from "./log"
+import { Location } from "./styled";
+import DERPLog from "./log";
 
 function DERP() {
   const [log, setLog] = useState([]);
   const [ip, setIp] = useState("-");
   const [country, setCountry] = useState("-");
-//  const [status, setStatus] = useState("not connected");
+  //  const [status, setStatus] = useState("not connected");
   const [rpcUrl, setRpcUrl2] = useState("-");
+  const [calls, setCalls] = useState(0);
+  const [currentTime, setCurrentTime] = useState(null);
+  const [elapsedTime, setElapsedTime] = useState(0);
   const [chainId, setChainId] = useState("-");
+  const [counter, setCounter] = useState("-");
   const [name, setName] = useState("-");
   const [city, setCity] = useState("-");
   const [coordinates, setCoordinates] = useState({
     long: undefined,
     lat: undefined,
   });
 
+  const startTimeRef = useRef(null);
+
   useEffect(() => {
-    console.log('rerendered DERP');
+    console.log("rerendered DERP");
   }, []);
 
   useEffect(() => {
-    console.log('coordinates', coordinates)
+    console.log("coordinates", coordinates);
   }, [coordinates]);
 
+  useEffect(() => {
+    if (calls > 1) {
+      if (!startTimeRef.current) {
+        startTimeRef.current = Date.now();
+      }
+      setCurrentTime(Date.now());
+      const intervalId = setInterval(() => {
+        setCurrentTime(Date.now());
+      }, 1000);
+
+
+      return () => clearInterval(intervalId);
+    }
+  }, [calls]);
+
+  useEffect(() => {
+    if (startTimeRef.current  && currentTime) {
+      setElapsedTime((currentTime - startTimeRef.current) / 1000);
+
+      const minutes = Math.floor(elapsedTime / 60);
+      const seconds = Math.floor(elapsedTime % 60);
+
+      setCounter(
+        `${calls} calls, ${minutes > 9 ? minutes : `0${minutes}`}:${
+          seconds > 9 ? seconds : `0${seconds}`
+        } running`
+      );
+
+    }
+  }, [elapsedTime, currentTime, calls]);
+
+  useEffect(() => {
+    if (calls > 1) {
+      if (!startTimeRef.current) {
+        startTimeRef.current = Date.now();
+      }
+      setCurrentTime(Date.now());
+      const intervalId = setInterval(() => {
+        setCurrentTime(Date.now());
+      }, 1000);
+
+
+      return () => clearInterval(intervalId);
+    }
```

Comment:
Won't this part of the code will be called every time the `calls` is incremented and bigger than 1? 
The interval constructor should be called only once